moderation/commands.go

```diff
@@ -602,6 +606,150 @@ var ModerationCommands = []*commands.YAGCommand{
 			return fmt.Sprintf("Deleted %d warnings.", rows), nil
 		},
 	},
+	&commands.YAGCommand{
+		CustomEnabled: true,
+		CmdCategory:   commands.CategoryModeration,
+		Name:          "ListViolationsCount",
+		Description:   "Lists Violations summary in entire server or of specified user optionally filtered by max violation age.\n Specify number of violations to skip while fetching using -skip flag ; max entries fetched 500",
+		Aliases:       []string{"ViolationsCount", "VCount"},
+		RequiredArgs:  0,
+		Arguments: []*dcmd.ArgDef{
+			&dcmd.ArgDef{Name: "User", Type: dcmd.UserID},
+		},
+		ArgSwitches: []*dcmd.ArgDef{
+			&dcmd.ArgDef{Switch: "ma", Name: "Max Violation Age", Default: time.Duration(0), Type: &commands.DurationArg{}},
+			&dcmd.ArgDef{Switch: "skip", Name: "Amount Skipped", Type: dcmd.Int, Default: 0},
+		},
+		RunFunc: func(parsed *dcmd.Data) (interface{}, error) {
+			config, _, err := MBaseCmd(parsed, 0)
+			if err != nil {
+				return nil, err
+			}
+			
+			//Roles with warn permissions can also list Violations.
+			_, err = MBaseCmdSecond(parsed, "", true, discordgo.PermissionManageMessages, config.WarnCmdRoles, true)
+			if err != nil {
+				return nil, err
+			}
+
+			userID := parsed.Args[0].Int64()			
+			order:= "id desc"
+			limit:= 500
+
+			//Check Flags
+			maxAge := parsed.Switches["ma"].Value.(time.Duration)
+			skip := parsed.Switches["skip"].Int()
+			if skip < 0 {
+				skip = 0
+			}
+			
+			// retrieve Violations
+			qms := []qm.QueryMod{qm.Where("guild_id = ?", parsed.GS.ID), qm.OrderBy(order), qm.Limit(limit), qm.Offset(skip)}
+			
+			if userID != 0 {
```

Comment:
can do the same with max age, example: `qm.Where("created_at > ?", time.Now().Add(-maxAge))`

Reply:
Yeah that looks like a much better and more consistent method. Made the changes. I plan to make some changes to the pagination method to not throw an error(Something went wrong when running this command, either discord or the bot may be having issues.) in first run when there are no results but ill put that as a separate PR later. This seems to be the case with some other commands like automod logs as well.